docs/adr/adr-009-public-api.md

```diff
@@ -95,10 +95,12 @@ type HeaderModule interface {
 LocalHead(ctx context.Context) (*header.ExtendedHeader, error)
 // GetByHash returns the header of the given hash from the node's header store.
 GetByHash(ctx context.Context, hash tmbytes.HexBytes) (*header.ExtendedHeader, error)
-// GetByHeight returns the header of the given height from the node's header store.
-// If the header of the given height is not yet available, the request will hang
-// until it becomes available in the node's header store.
+// GetByHeight returns the header of the given height if it is available as either
+// the syncer's head or as the chain head of the node's header store. 
```

Comment:
The Syncer head and chain head might be too contextual. I think we should say it return the currently synced headers.
In the future, any header >= networkhead